controller/common/node_info.py

```diff
@@ -12,71 +14,57 @@ def __init__(self, node_id):
         node_name, nvme_nqn, fc_wwns_str, iscsi_iqn = utils.get_node_id_info(node_id)
         fc_wwns = fc_wwns_str.split(config.PARAMETERS_FC_WWN_DELIMITER)
         self.node_name = node_name
-        self.initiators = Initiators(nvme_nqn.strip(), fc_wwns, iscsi_iqn.strip())
+        self.initiators = Initiators([nvme_nqn], fc_wwns, [iscsi_iqn])
 
 
+@dataclass
 class Initiators:
     """
     Object containing node initiators (e.g. iqn, fc_wwns)
     """
+    nvme_nqns: list = field(default_factory=list)
+    fc_wwns: list = field(default_factory=list)
+    iscsi_iqns: list = field(default_factory=list)
 
-    def __init__(self, nvme_nqn, fc_wwns, iscsi_iqn):
-        """
-        Args:
-            nvme_nqn: nqn
-            fc_wwns : list of fc wwns
-            iscsi_iqn : iqn
-        """
-        self.nvme_nqn = nvme_nqn
-        self.fc_wwns = list(filter(None, fc_wwns))
-        self.iscsi_iqn = iscsi_iqn
-        self._nvme_nqn_lowercase = nvme_nqn.lower()
-        self._fc_wwns_lowercase_set = set(wwn.lower() for wwn in fc_wwns)
-        self._iscsi_iqn_lowercase = iscsi_iqn.lower()
-
-    def is_array_wwns_match(self, host_wwns):
-        """
-        Args:
-           host_wwns : storage host wwns list
+    def __post_init__(self):
+        self.nvme_nqns = self._filter_empty_parts(self.nvme_nqns)
+        self.fc_wwns = self._filter_empty_parts(self.fc_wwns)
+        self.iscsi_iqns = self._filter_empty_parts(self.iscsi_iqns)
 
-        Returns:
-           Is current host wwns matches
-        """
-        host_wwns_lower = [wwn.lower() for wwn in host_wwns]
-        return not self._fc_wwns_lowercase_set.isdisjoint(host_wwns_lower)
-
-    def is_array_iscsi_iqns_match(self, host_iqns):
-        """
-        Args:
-           host_iqns: storage host iqns list
-
-        Returns:
-           Is current host iqns matches
-        """
-        host_iqns_lower = [iqn.lower() for iqn in host_iqns]
-        return self._iscsi_iqn_lowercase in host_iqns_lower
-
-    def is_array_nvme_nqn_match(self, host_nqns):
-        """
-        Args:
-           host_nqns: storage host nqns list
-
-        Returns:
-           Is current host nqns matches
-        """
-        host_nqns_lower = [nqn.lower() for nqn in host_nqns]
-        return self._nvme_nqn_lowercase in host_nqns_lower
+    def _filter_empty_parts(self, ports):
+        ports_strip = [port.strip() for port in ports]
+        ports_filter = filter(None, ports_strip)
+        return list(ports_filter)
 
     def _get_iter(self):
-        if self.nvme_nqn:
-            yield array_config.NVME_OVER_FC_CONNECTIVITY_TYPE, self.nvme_nqn
+        for nvme_nqn in self.nvme_nqns:
+            yield array_config.NVME_OVER_FC_CONNECTIVITY_TYPE, nvme_nqn
         for fc_wwn in self.fc_wwns:
             yield array_config.FC_CONNECTIVITY_TYPE, fc_wwn
-        if self.iscsi_iqn:
-            yield array_config.ISCSI_CONNECTIVITY_TYPE, self.iscsi_iqn
+        for iscsi_iqn in self.nvme_nqns:
```

Comment:
I don't think so

Reply:
could you also cover this in the unit tests?

Reply:
done